code/modules/food/drywet.dm

```diff
@@ -0,0 +1,209 @@
+			return TRUE
+	return ..()
+
+/obj/item/reagent_containers/drywet/afterattack(var/obj/target, var/mob/user, var/flag)
+	if(!flag)
+		return
+	for(var/type in can_be_placed_into)
+		if(istype(target, type))
+			return
+	if(standard_pour_into(user, target))
+		return 1
+	if(standard_dispenser_refill(user, target))
+		return 1
+
+/obj/item/reagent_containers/drywet/attack(mob/M as mob, mob/user as mob, def_zone)
+	if(force && !(flags & NOBLUDGEON) && user.a_intent == I_HURT)
+		return ..()
+
+	if(standard_feed_mob(user, M))
+		return
+
+	return FALSE
+
+/obj/item/reagent_containers/drywet/self_feed_message(mob/user)
+	to_chat(user, SPAN_NOTICE("You drink from \the [src]"))
+/obj/item/reagent_containers/drywet/other_feed_message_start(mob/user, mob/target)
+	user.visible_message(SPAN_WARNING("[user] is trying to make [target] drink from \the [src]!"))
+
+/obj/item/reagent_containers/drywet/other_feed_message_finish(mob/user, mob/target)
+	user.visible_message(SPAN_WARNING("[user] has made [target] drink from \the [src]!"))
+
+//copied from obj/item/storage
+/obj/item/reagent_containers/drywet/proc/handle_item_insertion(obj/item/W, prevent_warning = 0)
+	if (!istype(W)) return 0
+	if (usr)
+		usr.prepare_for_slotmove(W)
+		usr.update_icons() //update our overlays
+
+	W.loc = src
+	W.on_enter_storage(src)
+	var/volume_taken = W.get_storage_cost() ** 2
+	solids[W] = volume_taken
+	untaken_capacity -=  volume_taken // checked for validity in can be inserted proc but dangerous if that proc is skipped
+
+	if(usr)
+		if (usr.client)
+			usr.client.screen -= W
+		W.dropped(usr)
+		add_fingerprint(usr)
+
+		if (!prevent_warning)
+			for (var/mob/M in viewers(usr, null))
+				if (M == usr)
+					to_chat(usr, SPAN_NOTICE("You put \the [W] into [src]."))
+				else if (M in range(1)) //If someone is standing close enough, they can tell what it is...
+					M.show_message(SPAN_NOTICE("\The [usr] puts [W] into [src]."))
+				else if (W && W.w_class >= ITEM_SIZE_NORMAL) //Otherwise they can only see large or normal items from a distance...
+					M.show_message(SPAN_NOTICE("\The [usr] puts [W] into [src]."))
+
+
+	update_icon()
+	return 1
+
+// custom for drywet
+/obj/item/reagent_containers/drywet/proc/drywet_poured_into(obj/item/reagent_containers/W, mob/user)
+	if(untaken_capacity == 0)
+		to_chat(user, SPAN_NOTICE("[src] is full."))
+		return TRUE
+
+	if(!W.reagents.total_volume)
+		to_chat(user, SPAN_NOTICE("[W] is empty."))
+		return TRUE // if it returns false, it drains from its target when empty
+
+	var/cantrans = min(W.amount_per_transfer_from_this, untaken_capacity)
+	var/trans = W.reagents.trans_to(src, cantrans)
+	untaken_capacity -= cantrans
+	to_chat(user, SPAN_NOTICE("You transfer [trans] units of the solution to [src]."))
+	playsound(src,'sound/effects/Liquid_transfer_mono.ogg',50,1)
+	user.investigate_log("transfered [trans] units from [src]([reagents.log_list()]) to [src]([reagents.log_list()])", "chemistry")
```

Comment:
```suggestion
	user.investigate_log("transfered [trans] units from [W]([reagents.log_list()]) to [src]([reagents.log_list()])", "chemistry")
```

Reply:
Should't the log be placed before the transfer happens to have the proper reagents in the brackets?

Reply:
normal reagent containers use the order used here.